servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/ConcurrentUtils.java

```diff
@@ -78,6 +79,36 @@ private ConcurrentUtils() {
         return lockUpdater.getAndSet(owner, CONCURRENT_IDLE) == CONCURRENT_EMITTING;
     }
 
+    /**
+     * Acquire a lock that is exclusively held with no re-entry, but attempts to acquire the lock while it is
+     * held can be detected by {@link #releaseLock(AtomicInteger)}.
+     * @param lock The {@link AtomicInteger} used to control the lock state.
+     * @return {@code true} if the lock was acquired, {@code false} otherwise.
+     */
+    public static boolean tryAcquireLock(AtomicInteger lock) {
+        for (;;) {
+            final int prevEmitting = lock.get();
```

Comment:
Perhaps`get()` → `intValue()`. When I see `get()` I always think "reference".

Reply:
I'll just leave it as `get()` bcz that is the accessor for `Atomic*` types